gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java

```diff
@@ -177,7 +177,7 @@ public PlanEntity create(final ExecutionContext executionContext, NewPlanEntity
             String id = newPlan.getId() != null && UUID.fromString(newPlan.getId()) != null ? newPlan.getId() : UuidString.generateRandom();
 
             newPlan.setId(id);
-            Plan plan = planConverter.toPlan(newPlan, getApiDefinitionVersion(api));
+            Plan plan = planConverter.toPlan(newPlan, getApiDefinitionVersion(api), api.getType());
```

Comment:
I believe this service is uniquely non-v4. (The v4 one is in the `v4` folder). So in theory there should never be a need for the api type, right?

Reply:
Yes, good point.